handlers/spendMsgHandler.go

```diff
@@ -16,43 +15,40 @@ type NextTxIndex func() uint16
 // FeeUpdater updates the aggregate fee amount in a block
 type FeeUpdater func(amt *big.Int) sdk.Error
 
-func NewSpendHandler(utxoStore store.UTXOStore, plasmaStore store.PlasmaStore, nextTxIndex NextTxIndex, feeUpdater FeeUpdater) sdk.Handler {
+func NewSpendHandler(outputStore store.OutputStore, blockStore store.BlockStore, nextTxIndex NextTxIndex, feeUpdater FeeUpdater) sdk.Handler {
 	return func(ctx sdk.Context, msg sdk.Msg) sdk.Result {
 		spendMsg, ok := msg.(msgs.SpendMsg)
 		if !ok {
 			panic("Msg does not implement SpendMsg")
 		}
 
 		txIndex := nextTxIndex()
-		blockHeight := plasmaStore.NextPlasmaBlockNum(ctx)
+		blockHeight := blockStore.NextPlasmaBlockNum(ctx)
 
 		// construct the confirmation hash
 		merkleHash := spendMsg.MerkleHash()
 		header := ctx.BlockHeader().DataHash
 		confirmationHash := sha256.Sum256(append(merkleHash, header...))
 
-		// positional part of these keys addeded when the new positions are created
-		var spenderKeys [][]byte
-		var positions []plasma.Position
-		positions = append(positions, plasma.NewPosition(blockHeight, txIndex, 0, nil))
-		spenderKeys = append(spenderKeys, append(spendMsg.Output0.Owner[:], positions[0].Bytes()...))
-		if spendMsg.HasSecondOutput() {
-			positions = append(positions, plasma.NewPosition(blockHeight, txIndex, 1, nil))
-			spenderKeys = append(spenderKeys, append(spendMsg.Output1.Owner[:], positions[1].Bytes()...))
+		/* Store Transaction */
+		tx := store.Transaction{
+			Transaction:      spendMsg.Transaction,
+			Spent:            make([]bool, len(spendMsg.Outputs)),
+			Spenders:         make([][]byte, len(spendMsg.Outputs)),
+			ConfirmationHash: confirmationHash[:],
+			Position:         plasma.NewPosition(blockHeight, txIndex, 0, big.NewInt(0)),
 		}
+		outputStore.StoreTx(ctx, tx)
```

Comment:
Given that you're also storing tx's here maybe call this `dataStore` instead of `outputStore`

Reply:
Hmm `dataStore` is a bit too vague since it doesn't include all sidechain data such as plasma blocks. Txs are essentially just a list of spent/unspent outputs with metadata. We can discuss this on discord

Reply:
This is something we can maybe open an issue for change in a different pr if we want. Doesn't take  too long to update the name change, but is nice to keep out of this pr since merge conflicts with develop and hamdis branch are already a pain